cmd/process-agent/main_common.go

```diff
@@ -294,9 +300,15 @@ func runAgent(exit chan struct{}) {
 		return
 	}
 
+	expVarPort := ddconfig.Datadog.GetInt("process_config.expvar_port")
+	if expVarPort <= 0 {
+		log.Criticalf("Invalid process_config.expvar_port -- %d, exiting", expVarPort)
+		cleanupAndExit(1)
+	}
```

Comment:
nit: I think instead exiting here we could log the issue as a `WARN` and fall back to the `DefaultProcessExpVarPort`. wdyt?

Reply:
I've thought the same way to be honest. However, with the current behavior `LoadProcessYamlConfig` will return an error:
https://github.com/DataDog/datadog-agent/blob/46ea49c0eedea005df5d6d614d49f808a3074365/pkg/process/config/yaml_config.go#L106-L108

... making `NewAgentConfig` return an error
https://github.com/DataDog/datadog-agent/blob/256c7257a616cfba0c896b357e4a0aa1341e7eeb/pkg/process/config/config.go#L260-L262

which makes the process-agent exit:
https://github.com/DataDog/datadog-agent/blob/0fdb9e013e7d72dfbb7f42d52b5f2635523f5378/cmd/process-agent/main_common.go#L210-L214

Do you think that it would be better to change the current behavior?

Reply:
Yeah, I think it's better to not fail in this case as it's a recoverable config error given we have a known default. Also, this is a secondary functionality for agent telemetry and not a core feature.

Reply:
You're right. I've updated the PR to make process-agent fallback to the default value for `expvar_port` and `cmd_port` if the user is settings an invalid value!